libethcore/Farm.h

```diff
@@ -259,21 +259,34 @@ class Farm: public FarmFace
 			if (hwmon) {
 				HwMonitorInfo hwInfo = i->hwmonInfo();
 				HwMonitor hw;
-				unsigned int tempC = 0, fanpcnt = 0;
+				hw.powerW = 0.0;
+				unsigned int tempC = 0, fanpcnt = 0, powerW = 0;
 				if (hwInfo.deviceIndex >= 0) {
 					if (hwInfo.deviceType == HwMonitorInfoType::NVIDIA && nvmlh) {
 						wrap_nvml_get_tempC(nvmlh, hwInfo.deviceIndex, &tempC);
 						wrap_nvml_get_fanpcnt(nvmlh, hwInfo.deviceIndex, &fanpcnt);
+						if(power) {
+							wrap_nvml_get_power_usage(nvmlh, hwInfo.deviceIndex, &powerW);
+							hw.powerW = powerW/((double)1000.0);
+						}
 					}
 					else if (hwInfo.deviceType == HwMonitorInfoType::AMD && adlh) {
 						wrap_adl_get_tempC(adlh, hwInfo.deviceIndex, &tempC);
 						wrap_adl_get_fanpcnt(adlh, hwInfo.deviceIndex, &fanpcnt);
+						if(power) {
+							wrap_adl_get_power_usage(adlh, hwInfo.deviceIndex, &powerW);
+							hw.powerW = powerW/((double)1000.0);
+						}
 					}
 #if defined(__linux)
 					// Overwrite with sysfs data if present
 					if (hwInfo.deviceType == HwMonitorInfoType::AMD && sysfsh) {
 						wrap_amdsysfs_get_tempC(sysfsh, hwInfo.deviceIndex, &tempC);
 						wrap_amdsysfs_get_fanpcnt(sysfsh, hwInfo.deviceIndex, &fanpcnt);
+						if(power) {
+							wrap_amdsysfs_get_power_usage(sysfsh, hwInfo.deviceIndex, &powerW);
+							hw.powerW = powerW/((double)1000.0);
```

Comment:
can't this line be used only once, since we now have milliwatt everywhere?
After that all looks good.

Reply:
I'll change it on the API PR